src/chainlit/lc/callbacks.py

```diff
@@ -155,50 +149,66 @@ def consume_last_prompt(self):
         self.last_prompt = None
         return last_prompt
 
-    def get_message_params(self):
-        llm_settings = self.llm_settings
-
-        indent = len(self.sequence) if self.sequence else 0
+    def get_author(self):
+        if self.sequence:
+            return self.sequence[-1].author
+        return config.ui.name
 
+    def get_last_message(self):
         if self.sequence:
-            author = self.sequence[-1]
-        else:
-            author = config.ui.name
+            return self.sequence[-1]
+        return self.emitter.session.last_message
```

Comment:
i think we should store the default last message when the callback is instantiated (using `self.emitter.session.last_message`).

If `self.sequence` becomes empty after being not empty, then `self.emitter.session.last_message` will have changed

Reply:
Mhh, not if last_message only keeps track of the last user message as it currently does; let's discuss this here: https://github.com/Chainlit/chainlit/pull/182#discussion_r1267855310